stdlib/public/Concurrency/Actor.cpp

```diff
@@ -649,17 +649,14 @@ class alignas(sizeof(void *) * 2) ActiveActorStatus {
     PriorityShift = 0x8,
   };
 
-#if SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION && SWIFT_POINTER_IS_4_BYTES
-  uint32_t Flags;
-  dispatch_lock_t DrainLock;
-  uint32_t Unused;
-#elif SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION && SWIFT_POINTER_IS_8_BYTES
   uint32_t Flags;
+#if SWIFT_CONCURRENCY_ENABLE_PRIORITY_ESCALATION
```

Comment:
The warnings were:
```
/Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/swift/stdlib/public/Concurrency/TaskPrivate.h:337:13: warning: constexpr constructor that does not initialize all members is a C++20 extension [-Wc++20-extensions]
...
/Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/swift/stdlib/public/Concurrency/Actor.cpp:688:13: warning: constexpr constructor that does not initialize all members is a C++20 extension [-Wc++20-extensions]
```
The reason I went with this approach is that it seemed unscalable to me to repeat the conditionals as originally written throughout the files to handle the `Unused` field in different locations. I'm happy to leave this part out of the PR if you'd like to address the warnings yourself, though.

Reply:
My C++ foo is weak, can we do `Unused = 0` in the declaration?

Reply:
Yeah, that's probably a better approach than I took. I'll turn these declarations into this:
```
LLVM_ATTRIBUTE_UNUSED uint32_t Unused = {};
```
I do wonder if there's maybe there's a way to use alignment attributes to create the effective padding that this unused member achieves without having to explicitly declare a member, but this should be fine.

Reply:
> I do wonder if maybe there's a way to use alignment attributes to create the effective padding that this unused member achieves without having to explicitly declare a member, but this should be fine.

Definitely worthwhile but I do find it useful to have an explicit member to increase visibility of this so that future me/anyone who looks at this, knows we have extra bits available here.